pgjdbc/src/test/java/org/postgresql/util/ReaderInputStreamTest.java

```diff
@@ -182,6 +184,26 @@ public void malformedSingleCharArrayReadTest() throws IOException {
     read(is, 0xF0, 0xA0, 0x9C, 0x8E);
   }
 
+  @Test
+  public void readsSmallerThanBlockSizeTest() throws Exception {
+    final int BLOCK = 8 * 1024;
+    final int DATASIZE = BLOCK * 5 + 57;
+    final byte[] data = new byte[DATASIZE];
+    final byte[] buffer = new byte[BLOCK];
+
+    InputStreamReader isr = new InputStreamReader(new ByteArrayInputStream(data));
+    ReaderInputStream r = new ReaderInputStream(isr);
+
+    int read;
+    int total = 0;
+
+    while ((read = r.read(buffer, 0, BLOCK)) > -1) {
```

Comment:
I can replace it with N calls to read (in this case 6). I don't think that using a single (bigger) read would fail the test when the problem occurs.